rust/noosphere-cli/src/native/commands/serve.rs

```diff
@@ -1,25 +1,28 @@
 //! Concrete implementations of subcommands related to running a Noosphere
 //! gateway server
 
+use crate::native::workspace::Workspace;
 use anyhow::Result;
-
+use noosphere_gateway::{start_gateway, GatewayScope};
+use noosphere_storage::StorageConfig;
 use std::net::{IpAddr, TcpListener};
-
 use url::Url;
 
-use crate::native::workspace::Workspace;
-
-use noosphere_gateway::{start_gateway, GatewayScope};
-
 /// Start a Noosphere gateway server
 pub async fn serve(
     interface: IpAddr,
     port: u16,
     ipfs_api: Url,
     name_resolver_api: Url,
     cors_origin: Option<Url>,
-    workspace: &Workspace,
+    storage_config: Option<StorageConfig>,
+    workspace: &mut Workspace,
 ) -> Result<()> {
+    if let Some(storage_config) = storage_config {
+        workspace.release_sphere_context();
+        workspace.with_storage_config(storage_config);
```

Comment:
It seems like we pass this arg all the way in here just to invoke the configuration right away. Perhaps we are doing this configuration in the wrong place. Why shouldn't the CLI handle this configuration? Why wouldn't it be provided at workspace instantiation time?

Reply:
agreed, this should be handled in `invoke_cli`